core/src/main/java/org/elasticsearch/action/get/MultiGetResponse.java

```diff
@@ -149,6 +186,56 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
+    public static MultiGetResponse fromXContent(XContentParser parser) {
+        List<MultiGetItemResponse> items = new ArrayList<>();
+        try {
+            String currentFieldName = null;
+            for (Token token = parser.nextToken(); token != Token.END_OBJECT; token = parser.nextToken()) {
+                switch (token) {
+                    case FIELD_NAME:
+                        currentFieldName = parser.currentName();
+                        break;
+                    case START_ARRAY:
+                        if (Fields.DOCS.equals(currentFieldName) == false) {
+                            throw new ElasticsearchParseException("Unexpected field [" + currentFieldName + "]");
```

Comment:
I don't think that throwing exceptions here is a good idea. We should rather be lenient to ensure forward compatibility. Say we add another array at the same level of docs (not likely, but still...) we should not throw error if your own client receives such a response. Rather ignore it.